src/components/data-management/ProjectsList.jsx

```diff
@@ -1,54 +1,82 @@
-import React from 'react';
+import React, { useCallback, useRef, useEffect } from 'react';
 import { useSelector } from 'react-redux';
 import PropTypes from 'prop-types';
 import { Space, Skeleton } from 'antd';
-
+import { VariableSizeList as List } from 'react-window';
 import integrationTestConstants from 'utils/integrationTestConstants';
 import ProjectCard from './ProjectCard';
 
 // This makes sure that all the projects can be viewed properly inside the list
 // TODO : This has to be done properly in CSS
-const windowMargin = 130; // px
+const windowMargin = 130;// px
+
+const Row = ({
+  index, data, style, setSize,
+}) => {
+  const rowRef = useRef();
+  const experiment = data[index];
+
+  useEffect(() => {
+    if (rowRef.current) {
+      setSize(index, rowRef.current.getBoundingClientRect().height);
+    }
+  }, [setSize, index, experiment]);
+
+  return (
+    <Space style={{ ...style, width: '100%' }}>
+      <div ref={rowRef}>
+        <ProjectCard key={experiment.id} experimentId={experiment.id} />
+      </div>
+    </Space>
+  );
+};
 
 const ProjectsList = (props) => {
   const { height, filter } = props;
+  const listRef = useRef();
+  const sizeMap = useRef({});
 
   const experiments = useSelector((state) => state.experiments);
 
-  if (experiments.meta.loading) {
-    return [...Array(5)].map((_, idx) => <Skeleton key={`skeleton-${idx}`} role='progressbar' active />);
-  }
-
-  // if there are no element return an emtpy one so that the tests know the list has been loaded
-  if (experiments.ids.length === 0) {
-    return (
-      <div data-test-id={integrationTestConstants.ids.PROJECTS_LIST} />
-    );
-  }
+  const setSize = useCallback((index, size) => {
+    sizeMap.current = { ...sizeMap.current, [index]: size + 5 };
 
-  return (
-    <Space
-      data-test-id={integrationTestConstants.ids.PROJECTS_LIST}
-      direction='vertical'
-      style={{ height: height - windowMargin, overflowY: 'auto' }}
-    >
-      {
-        experiments.ids.map((experimentId) => {
-          const experiment = experiments[experimentId];
+    //  if the height gets changed, we need to reset the heights
+    // so that they are recalculated
+    listRef.current.resetAfterIndex(index);
+  }, []);
 
-          const matchFilter = experiment.name.match(filter) || experimentId.match(filter);
+  const getSize = (index) => sizeMap.current[index] || 204; // default height if not yet measured
```

Comment:
Why do we need a `sizeMap`? And why do we need to keep track of heights? Isn't it a set height all the time?

Reply:
Some cards can have a bigger height, depending on how long the title is. The heights need to be calculated for the virtualized list for each card.
https://codepen.io/dariuszpaluch/pen/QYdadM?editors=0110